src/test/scala/DBIntegrationSpec.scala

```diff
@@ -3,30 +3,38 @@ import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.time.{Millis, Seconds, Span}
 import org.scalatest.{Matchers, WordSpec}
 import slick.jdbc.PostgresProfile.api._
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
 
-import scala.concurrent.Future
+import scala.concurrent.duration._
 
 class DBIntegrationSpec extends WordSpec with Matchers with DockerTestKit with DockerPostgresSetup
   with ScalaFutures {
 
-  implicit val defaultPatience =
+  implicit val defaultPatience: PatienceConfig =
     PatienceConfig(timeout =  Span(20, Seconds), interval = Span(5, Millis))
 
   "DB connection" must {
     "be obtained from container" in new InitDockerDB {
-      private val eventualCoffees: Future[Seq[Coffee]] = for {
-        _       <- Future(Thread.sleep(2000))//need to wait for postgres to start in container
+      private val eventualCoffees: Task[Seq[Coffee]] = for {
+        _       <- postgresStarted
         _       <- runFlywayMigration
-        coffees <- db.run(Coffees.coffees.result.map {
-                      _.map {
-                        case (name, price) => Coffee(name, price)
-                      }
-                    })
+        coffees <- Task.deferFuture(db.run(Coffees.coffees.result.map {
+          _.map {
+            case (name, price) => Coffee(name, price)
+          }
+        }))
       } yield coffees
 
-      whenReady(eventualCoffees) { result =>
+      whenReady(eventualCoffees.runAsync) { result =>
         result.length should be > 0
       }
     }
   }
+
+  def postgresStarted: Task[Boolean] =
+    Task.deferFuture(isContainerReady(postgresContainer)).timeout(10 seconds)
+  def scheduleAfter(duration: FiniteDuration): Task[Unit] =
```

Comment:
Where is this used?

Reply:
ah, good catch, `scheduleAfter` is not need any longer, will remove